File: python/src/runapi/core/multipart.py
"""Multipart upload payloads."""

from dataclasses import dataclass
from typing import Any, Dict, Optional


@dataclass
class MultipartFile:
    path: str
    filename: Optional[str] = None
    content_type: Optional[str] = None


class MultipartBody:
    """A multipart/form-data request body of plain fields plus file parts."""

    def __init__(
        self,
        fields: Optional[Dict[str, Any]] = None,
        files: Optional[Dict[str, MultipartFile]] = None,
    ) -> None:
        self.fields: Dict[str, Any] = {str(key): value for key, value in (fields or {}).items()}
        self.files: Dict[str, MultipartFile] = {str(key): value for key, value in (files or {}).items()}

File: go/core/multipart.go
package core

// MultipartFile describes a file part in a multipart/form-data request.
type MultipartFile struct {
	// Path is the local filesystem path to the file. Required.
	Path string
	// FileName overrides the filename sent in the Content-Disposition header.
	// When empty, the base name of Path is used.
	FileName string
	// ContentType sets the MIME type of the file part. When empty, the
	// multipart writer's default (application/octet-stream) is used.
	ContentType string
}

// MultipartBody describes fields and file parts for a multipart/form-data request.
type MultipartBody struct {
	// Fields are plain text form fields keyed by field name.
	Fields map[string]string
	// Files are file attachments keyed by the form field name.
	Files map[string]MultipartFile
}
